docs/user-guide/petset.md

```diff
@@ -0,0 +1,278 @@
+
+The goal of Pet Set is to decouple this dependency by assigning identities to individual instances of an application that are not anchored to the underlying physical infrastructure. For the rest of this document we will refer to these entities as "Pets". Our use of this term is predated by the "Pets vs Cattle" analogy.
+
+__Relationship between Pets and Pods__: PetSet requires there be {0..N-1} Pets. Each Pet has a deterministic name - PetSetName-Ordinal, and a unique identity. Each Pet has at most one pod, and each Pet Set has at most one Pet with a given identity.
+
+## When to use Pet Set?
+
+A Pet Set ensures that a specified number of "pets" with unique identities are running at any given time. The identity of a Pet is comprised of:
+
+* a stable hostname, available in DNS
+* an ordinal index
+* stable storage: linked to the ordinal & hostname
+
+These properties are useful in deploying stateful applications. However most stateful applications are also clustered, meaning they form groups with strict membership requirements that rely on stored state. Pet Set also helps with the 2 most common problems encountered managing such clustered applications:
+
+* discovery of peers for quorum
+* startup/teardown ordering
+
+Only use Pet Set if your application requires some or all of these properties. Managing pods as stateless replicas is vastly easier.
+
+Example workloads for Pet Set:
+
+* Databases like MySQL or PostgreSQL that require a single instance attached to a NFS persistent volume at any time
+* Clustered software like Zookeeper, Etcd, or Elasticsearch that require stable membership.
+
+## Alpha limitations
+
+Before you start deploying applications as Pet Sets, there are a few limitations you should understand.
+
+* Pet Set is an *alpha* resource, not available in any Kubernetes release prior to 1.3.
+* As with all alpha/beta resources, it can be disable through the `--runtime-config` option passed to the apiserver, and in fact most likely will be disabled on hosted offerings of Kubernetes.
+* The only updatable field on a Pet Set is `replicas`
+* The storage for a given pet must either be provisioned by a [dynamic storage provisioner](http://releases.k8s.io/{{page.githubbranch}}/examples/experimental/persistent-volume-provisioning/README.md) based on the requested `storage class`, or pre-provisioned by an admin. Note that dynamic volume provisioning is also currently in alpha.
+* Deleting the Pet Set  *will not*  delete any pets. You will either have to manually scale it down to 0 pets first, or delete the pets yourself.
+* Deleting and/or scaling a Pet Set down will *not* delete the volumes associated with the Pet Set. This is done to ensure safety first, your data is more valuable than an auto purge of all related Pet Set resources. **Deleting the Persistent Volume Claims will result in a deletion of the associated volumes**.
```

Comment:
I considered including `also, rm -rf /, don't do that`, but decided to leave it out